AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -371,30 +371,58 @@ - (void)flushQueue {
 
 - (void)checkPendingLogs {
 
-  // Skip checking pending logs if the channel is paused.
-  if (self.paused) {
-    return;
-  }
-
   // If the interval is default and we reached batchSizeLimit flush logs now.
-  if (self.configuration.flushInterval == kMSFlushIntervalDefault && self.itemsCount >= self.configuration.batchSizeLimit) {
+  if (!self.paused && self.configuration.flushInterval == kMSFlushIntervalDefault && self.itemsCount >= self.configuration.batchSizeLimit) {
     [self flushQueue];
   } else if (self.itemsCount > 0) {
     NSUInteger flushInterval = [self resolveFlushInterval];
-    if (flushInterval == 0) {
 
-      // If the interval is over, send all logs without any additional timers.
+    // Skip sending logs if the channel is paused.
+    if (self.paused) {
+      return;
+    }
+
+    // If the interval is over, send all logs without any additional timers.
+    if (flushInterval == 0) {
       [self flushQueue];
-    } else {
+    }
 
-      // Postpone sending logs.
+    // Postpone sending logs.
+    else {
       [self startTimer:flushInterval];
     }
   }
 }
 
 #pragma mark - Timer
 
+- (NSUInteger)resolveFlushInterval {
```

Comment:
Consider adding a comment to this method, you can use the same one from iOS: `Calculate remaining interval to trigger ingestion based on initial batch interval and stored start value.`.

The reason I ask for a comment is I found the `resolveFlushInterval` method name confusing. Because I was under the impression that it's simply calculating flush time intervals, but it also storing the `oldestPendingLogTimestampKey`.

Reply:
We have a comment in the private header MSChannelUnitDefaultPrivate.h, should I move/copy it to .m file?

Reply:
> Chúng tôi có một nhận xét trong tiêu đề riêng MSChannelUnitDefaultPrivate.h, nó có thể truy cập như một tài liệu cho phương pháp này. Tôi có nên di chuyển / sao chép nó vào tập tin .m không?

Reply:
@russelarms, if it's already in private header MSChannelUnitDefaultPrivate.h, then we don't need to add it in .m file.